src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,104 @@
+}]
+
+# todo: oak issue: too slow iterating. 36 seconds for below snippet
+# for mondo_id in mondo_ids[0:10]:
+#     rels: List[CURIE] = [x for x in oi.relationships(subjects=[mondo_id])]
+
+# todo: add caching?
+def sync_subclassof(
+    outpath: str = DEFAULT_OUTPATH, prefixes_csv_path: str = DEFAULT_PREFIXES_CSV,
+    ontology_db_path: str = DEFAULT_ONTOLOGY_DB_PATH
+):
+    """Run"""
+    # Vars
+    conv = get_monarch_curies_converter(prefixes_csv_path)
+    compress = lambda x: conv.compress(x) if conv.compress(x) else x
+    oi = get_adapter(ontology_db_path)
+
+    # Get all terms
+    ids_all: List[Union[CURIE, URI]] = [x for x in oi.entities(filter_obsoletes=False)]
+    mondo_ids: List[CURIE] = [x for x in ids_all if x.startswith('MONDO:')]
+
+    # todo: pr comment?: skip if parent_id not in mondo? is this only the case for: [('MONDO:0000001', 'rdfs:subClassOf', 'BFO:0000016')]
+    #  TODO: analyze first
+    #   example: mondo:0000001 rdfs:subClassOf BFO:0000016
+    # Relationships
+    # todo: PR comment: how did i implement this and why. memory vs speed. timeout on all rels. 36 seconds on iter 1x/time for just 10 records
+    # todo: oak issue: slow even batch rels (all preds). 5 seconds for 10. 25 for 100. when passed 5 preds, 1 second for 100. 36 sec for 25k~
+    rels_raw: List[RELATIONSHIP] = [x for x in oi.relationships(
+        subjects=mondo_ids,
+        # todo: pr comment: correct rels? close/related too? I think for GARD we might have 1-2 related cuz unsure
+        #  - and another question: are we indeed putting all of these in same column? (currently "'xref_ids': 'A oboInOwl:hasDbXref SPLIT=|'")
+        predicates=['rdfs:subClassOf', 'oboInOwl:hasDbXref', 'skos:exactMatch', 'skos:broadMatch', 'skos:narrowMatch']
+    )]
+    rels: List[RELATIONSHIP] = [(x[0], x[1], compress(remove_angle_brackets(x[2]))) for x in rels_raw]
+
+    # Create rows
+    # - Collect rows using lists
+    # TODO: pr comment: want to filter out only certain ontologies? (e.g. snomed shows up but we have no ingest)
+    rows_by_id: Dict[CURIE, Dict[str, Any]] = {}
+    for rel in rels:
+        sub, pred, obj = rel
+        if sub not in rows_by_id:
+            rows_by_id[sub] = {
+                'term_id': sub,
+                'parent_id': [],
+                'xref_ids': [],
+            }
+        if pred == 'rdfs:subClassOf':
+            rows_by_id[sub]['parent_id'].append(obj)
+        else:
+            rows_by_id[sub]['xref_ids'].append(obj)
+    # - Convert lists to pipe-delimited strings
+    rows_by_id2: Dict[CURIE, Dict[str, Any]] = {}
+    for k, v in rows_by_id.items():
+        rows_by_id2[k] = {k2: '|'.join(v2) if isinstance(v2, list) else v2 for k2, v2 in v.items()}
+
+    df = pd.DataFrame(rows_by_id2.values()).sort_values(by=['term_id'])
+    out_df = pd.concat([pd.DataFrame(ROBOT_ROW), df])
+    out_df.to_csv(outpath, sep='\t', index=False)
+
+
+def cli():
```

Comment:
## Holding off on `click`
@matentzn You wanted me to use `click` in this PR, but I felt that the best thing to do was (at least at first) complete the main task. I also think that maybe it's best to do `click` as a separate PR; maybe do it for all the scripts at once, but that will take a bit of effort and involve @hrshdhgd's review. I have some existing reference material / boilerplate for that from @hrshdhgd's code, etc.

Do you still want me to use a `click` CLI for this PR? Or OK w/ handling that as a separate issue (#255)?

Reply:
OK, though I have [some disagreements](https://github.com/monarch-initiative/mondo-ingest/pull/363#discussion_r1332369923) with that approach.